### PR TITLE
feat: delete product

### PR DESCRIPTION
**Pull request by davidolanrewaju**
## What does the PR do?
This PR addresses issue #77 by implementing the functionality to delete a single product from the database.

#### To test this feature, you can send a DELETE request to the following endpoint:

``` DELETE /api/v1/products/{id} ``` - where {id} is the ID of the product you wish to delete.

### Success Response 
If the product is deleted successfully, you will receive the following response structure:
```
{
  "status_code": 200,
  "message": "Product deleted successfully",
}
```

### "Product Does Not Exist" Response
If the product doesn't exist in the database, the following response will be received:
```
{
	error: 'Product not found',
        message: 'The product with ID ${id} does not exist',
}
```

### Invalid Authentication Response
If the request is made by an unauthenticated or unauthorized user, the following response will be received:
```
{
	"error": "Invalid request",
	"message": "User not authenticated or authorized"
}
```

## Task
- [x] Implement authentication logic to ensure the client is signed in
- [x] Grant user the ability to delete registered products
- [x] Ensure the endpoint correctly identifies if the client is signed in
- [x] Test the endpoint correctly deletes any single product selected by client.

## Referenced Issue
[[FEAT] : Delete product API - Backend](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/77)

### CODE REVIEW

File: src/modules/products/entities/product.entity.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { AbstractBaseEntity } from '../../../entities/base.entity';
+import { User } from '../../user/entities/user.entity';
+import { ApiProperty } from '@nestjs/swagger';
+import { Column, Entity, ManyToOne } from 'typeorm';
+
+@Entity()
+export class Product extends AbstractBaseEntity {
+  @ApiProperty()
```
Comment (Cobidex) [inline src/modules/products/entities/product.entity.ts:8]: This decorator should be on the Dto which is the primary interface for exchanging data between the client and the API. It's not meant to be on the entity

**Comment by Cobidex** [comment]
Your tests are failing.

**Review by Cobidex — APPROVED** [review]

File: src/modules/products/entities/product.entity.ts
Comment (Heba-WebDev) [inline src/modules/products/entities/product.entity.ts:1]: I think you should specify if a column is not nullable, especially if you want to enforce this constraint at the database level.

**Review by Heba-WebDev — CHANGES_REQUESTED** [review]
Please update the dto, the rest looks good to me

**Review by Heba-WebDev — APPROVED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Pull request closed without merging**